wire/internal/wire/analyze.go

```diff
@@ -433,15 +430,15 @@ func verifyAcyclic(providerMap *typeutil.Map, hasher typeutil.Hasher) []error {
 
 // bindingConflictError creates a new error describing multiple bindings
 // for the same output type.
-func bindingConflictError(fset *token.FileSet, pos token.Pos, typ types.Type, prevSet *ProviderSet) error {
-	typString := types.TypeString(typ, nil)
-	var err error
-	if prevSet.VarName == "" {
-		err = fmt.Errorf("multiple bindings for %s (previous binding at %v)",
-			typString, fset.Position(prevSet.Pos))
+func bindingConflictError(fset *token.FileSet, typ types.Type, set *ProviderSet, cur, prev *providerSetSrc) error {
+	sb := new(strings.Builder)
+	if set.VarName == "" {
+		fmt.Fprintf(sb, "wire.Build")
 	} else {
-		err = fmt.Errorf("multiple bindings for %s (previous binding in %q.%s)",
-			typString, prevSet.PkgPath, prevSet.VarName)
+		fmt.Fprintf(sb, set.VarName)
 	}
-	return notePosition(fset.Position(pos), err)
+	fmt.Fprintf(sb, " has multiple bindings for %s (", types.TypeString(typ, nil))
+	fmt.Fprintf(sb, "current binding: %s", strings.Join(cur.trace(fset, typ), " <- "))
```

Comment:
I considered using multi-line errors here (e.g., replacing " <- " with "\n  " or some such), but don't think the test framework supports that. Maybe a TODO?

Reply:
Oh. Yes, the test compares against the whole error string (not line-based), but you can't specify a match over multiple lines. One possible route could be to add something like a trailing backslash to indicate line continuations. A TODO seems fine.

Reply:
I filed an issue to track this.